cmd/apply-crds/main.go

```diff
@@ -16,170 +16,8 @@ limitations under the License.
 
 package main
 
-import (
-	"context"
-	"flag"
-	"fmt"
-	"io"
-	"log"
-	"os"
-	"path/filepath"
-	"strings"
-
-	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
-	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
-	v1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/apimachinery/pkg/util/yaml"
-	"k8s.io/client-go/util/retry"
-	ctrl "sigs.k8s.io/controller-runtime"
-)
-
-type StringList []string
-
-func (s *StringList) String() string {
-	return strings.Join(*s, ", ")
-}
-
-func (s *StringList) Set(value string) error {
-	*s = append(*s, value)
-	return nil
-}
-
-var (
-	crdsDir StringList
-)
-
-func initFlags() {
-	flag.Var(&crdsDir, "crds-dir", "Path to the directory containing the CRD manifests")
-	flag.Parse()
-
-	if len(crdsDir) == 0 {
-		log.Fatalf("CRDs directory is required")
-	}
-
-	for _, crdDir := range crdsDir {
-		if _, err := os.Stat(crdDir); os.IsNotExist(err) {
-			log.Fatalf("CRDs directory %s does not exist", crdsDir)
-		}
-	}
-}
+import "github.com/NVIDIA/k8s-operator-libs/pkg/applycrds"
```

Comment:
Can we think of package name that is aligned with the Effective Go guidelines?

As per this [page](https://go.dev/blog/package-names), good package names are often nouns. In this case, I was thinking a package name like `crds` or `kubeclient/crds` could do the trick. What are your thoughts @tobiasgiese ?

Reply:
That is a good call. What do you think about a package name `crdutil`? If we rename the func `Run()` to `EnsureCRDsCmd()` we are even able to add more commands related to crds -- if ever needed.

```diff
diff --git a/pkg/applycrds/applycrds.go b/pkg/crdutil/crdutil.go
similarity index 97%
rename from pkg/applycrds/applycrds.go
rename to pkg/crdutil/crdutil.go
index 7a16217..9fa4f9e 100644
--- a/pkg/applycrds/applycrds.go
+++ b/pkg/crdutil/crdutil.go
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */

-package applycrds
+package crdutil

 import (
 	"context"
@@ -67,9 +67,9 @@ func initFlags() {
 	}
 }

-// Run reads each YAML file in the directory, splits it into documents, and applies each CRD to the cluster.
+// EnsureCRDsCmd reads each YAML file in the directory, splits it into documents, and applies each CRD to the cluster.
 // The parameter --crds-dir is required and should point to the directory containing the CRD manifests.
-func Run() {
+func EnsureCRDsCmd() {
 	ctx := context.Background()

 	initFlags()
```